src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -884,16 +936,34 @@ class SceneryWindow final : public Window
             }
         }
 
-        // Remove misc tab if empty
-        if (_tabEntries.back().Entries.size() == 0)
+        // Remove empty tabs
+        for (auto it = _tabEntries.begin(); it != _tabEntries.end();)
+        {
+            if (it->Entries.size() == 0)
+                it = _tabEntries.erase(it);
+            else
+                it++;
+        }
+
+        // Move all scenery tab to end of first row
```

Comment:
This sounds like a std::rotate.

Reply:
this should look better now, thanks for the tip